hotsos/cli.py

```diff
@@ -303,49 +345,53 @@ def cli(data_root, version, defs_path, templates_path, all_logs, debug,
+
         _version = get_version()
-        if version:
+        if arguments.version:
             print(_version)
             return
 
         config = {'repo_info': get_repo_info(),
-                  'force_mode': force,
+                  'force_mode': arguments.force,
                   'hotsos_version': _version,
-                  'command_timeout': command_timeout,
-                  'use_all_logs': all_logs,
-                  'plugin_yaml_defs': defs_path,
-                  'templates_path': templates_path,
-                  'event_tally_granularity': event_tally_granularity,
-                  'max_logrotate_depth': max_logrotate_depth,
-                  'max_parallel_tasks': max_parallel_tasks,
-                  'machine_readable': machine_readable,
-                  'debug_mode': debug,
-                  'scenario_filter': scenario,
-                  'event_filter': event}
+                  'command_timeout': arguments.command_timeout,
+                  'use_all_logs': arguments.all_logs,
+                  'plugin_yaml_defs': arguments.defs_path,
+                  'templates_path': arguments.templates_path,
+                  'event_tally_granularity': arguments.event_tally_granularity,
+                  'max_logrotate_depth': arguments.max_logrotate_depth,
+                  'max_parallel_tasks': arguments.max_parallel_tasks,
+                  'machine_readable': arguments.machine_readable,
+                  'debug_mode': arguments.debug,
+                  'scenario_filter': arguments.scenario,
+                  'event_filter': arguments.event}
         HotSOSConfig.set(**config)
 
         with LoggingManager() as logmanager:
-            with DataRootManager(data_root,
-                                 sos_unpack_dir=sos_unpack_dir) as drm:
+            with DataRootManager(
+                arguments.data_root, sos_unpack_dir=arguments.sos_unpack_dir
+            ) as drm:
                 HotSOSConfig.data_root = drm.data_root
                 if is_snap() and drm.data_root == '/':
                     print(SNAP_ERROR_MSG)
                     sys.exit(1)
 
-                if debug and quiet:
+                if arguments.debug and arguments.quiet:
                     sys.stderr.write('ERROR: cannot use both --debug and '
                                      '--quiet\n')
                     return
 
                 # Set a name so that logs have this until real plugins are run.
                 log.name = 'hotsos.cli'
 
-                if list_plugins:
+                if arguments.list_plugins:
                     sys.stdout.write('\n'.join(plugintools.PLUGINS.keys()))
                     sys.stdout.write('\n')
                     return
 
-                with progress_spinner(not quiet and not debug, drm.name):
+                with progress_spinner(
+                        not arguments.quiet and not arguments.debug, drm.name):
                     plugins = []
```

Comment:
This is now unfortunately broken. we were using kwargs that were not explicitly defined i.e. from  (**kwargs) as plugin names to be run and since almost everything is now in kwargs this now blows up when you use any on the command line. We therefore need to find a way to identify plugin names in kwargs - maybe could pull the nested new_init() function out of filter_unexpected_field() and generalise it to use for this purpose also.

Reply:
Good catch, I'll fix it shortly

Reply:
Should be good to go.